src/components/renderer/SvgRenderer.tsx

```diff
@@ -0,0 +1,182 @@
+import { v2, utils } from "@govtechsg/open-attestation";
+import React, { CSSProperties, FunctionComponent, useEffect, useState } from "react";
+import { renderToStaticMarkup } from "react-dom/server";
+import crypto from "crypto";
+import bs58 from "bs58";
+import { ConnectionFailureTemplate, NoTemplate, TamperedSvgTemplate } from "../../DefaultTemplate";
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const handlebars = require("handlebars");
+
+interface SvgRendererProps {
+  document: any; // TODO: Update to OpenAttestationDocument
+  svgRef: React.RefObject<HTMLIFrameElement>;
+  svgData?: string;
+  style?: CSSProperties;
+  className?: string;
+  sandbox?: string;
+  onConnected?: () => void; // Optional call method to call once svg is loaded
+  forceV2?: boolean;
+}
+
+const EMBEDDED_DOCUMENT = "[Embedded SVG]";
+
+enum DisplayResult {
+  OK = 0,
+  DEFAULT = 1,
+  CONNECTION_ERROR = 2,
+  DIGEST_ERROR = 3,
+}
+
+export const SvgRenderer: FunctionComponent<SvgRendererProps> = ({
+  document,
+  svgRef,
+  svgData,
+  style,
+  className,
+  sandbox = "allow-same-origin",
+  onConnected,
+  forceV2 = false,
+}) => {
+  const [buffer, setBuffer] = useState<ArrayBuffer>();
+  const [svgFetchedData, setFetchedSvgData] = useState<string>("");
+  const [source, setSource] = useState<string>("");
+  const [toDisplay, setToDisplay] = useState<DisplayResult>(DisplayResult.OK);
+
+  let docAsAny: any;
+  if (forceV2 && utils.isRawV2Document(docAsAny)) {
```

Comment:
I think you mean to check `document`?

Reply:
Removed this portion entirely due to the new logic